src/toil/jobStores/aws/jobStore.py

```diff
@@ -299,6 +303,65 @@ def deleteFile(self, jobStoreFileID):
         else:
             info.delete()
 
+    def importFile(self, sourceUrl):
+        url = urlparse.urlparse(sourceUrl)
+        if url.scheme.lower() == 's3':
+            if len(sourceUrl.split('/')) != 4:
```

Comment:
note also that the regex's should be precompiled and stored in a higher level class variable above the import and export functions so that they do not need to be compiled on each function call.